### ISSUE TITLE
Maps from projective schemes to R

### ISSUE THREAD

**Issue by felixwellen** [felixwellen/synthetic-zariski#6]
I remember that @fabianmasato and @mnieper  discussed a proof with me, that maps from a connected projective scheme, i.e. a connected closed subset of some $\mathbb{P}^n$, to the base ring $R$ should be constant. Does anyone remember the details?

**Comment by dwarn** [comment]
This is false, since for example any infinitesimal disk is a connected projective scheme and admits a non-constant embedding in R. What's true is that such a map cannot assume distinct values. By "compactness", for any map f: X -> R with X projective, the complement of the image of f is open in R, i.e. given by an ideal I in R[X]. If X is non-empty then this ideal is non-zero. Up to not-not we can find a deduplicated set of roots of some non-zero element p of I. Say p has distinct roots a1 ... ak. Now f(x) is not in the open determined by I, so p(f(x)) is not not zero, which means there is a unique i such that f(x) is not not ai. If X is connected then this i is the same for all x : X, which proves the claim.

**Comment by felixwellen** [comment]
> If X is non-empty then this ideal is non-zero.

I'm confused: Shouldn't X non-empty rather imply something like $I$ is not $(1)$?

But other than that, I can follow. So in general one could say: Whenever we have a connected $X$ and $f:X\to R$ such that $\mathrm{im}(f)\subseteq V(g)$, with $g:R[X]$ of degree at least 1, then $\mathrm{im}(f)\subseteq \mathbb{D}(f(\ast))$.

**Comment by dwarn** [comment]
> I'm confused: Shouldn't X non-empty rather imply something like $I$ is not $(1)$?

Yes, of course. I think the reason $I$ is non-zero is if you consider the composition $X \to \mathbb A^1 \hookrightarrow \mathbb P^1$, again the complement of the image of this composition is open. So we have an open subset of $\mathbb P^1$ containing the point at infinity. Such an open subset cannot have empty intersection with $\mathbb A^1$. So also the complement of the image of the map $X \to \mathbb A^1$ is non-empty.

**Comment by felixwellen** [comment]
Ah. I think the same trick was used for the theorem I didn't remember ;-)
So, fortunately in this case, I guess we can consistently define the complement $C([x:y])$ as $(z:X)\to x\cdot f(z)\neq y$.

**Comment by felixwellen** [comment]
(Or at least some non-empty open set with empty intersection with the image of f, which is enough)

**Comment by felixwellen** [comment]
So for a projective affine connected pointed type $X$ we should have that the type of maps $X\to R$ is a quotient of a finite free R-module. By the above, we have $\mathrm{im}(f)\subseteq \mathbb{D}(f(\ast))$, for any map $f:X\to R$. Then $f$ merely factors over a standard disk $D(n)$ for some $n$. Since $X$ is affine, the type of maps $X\to R$ is finitely generated by $f_1,\dots,f_k$. We can take a maximal $n$, such that all $f_i$ factor over $D(n)$. This means there is a surjective algebra homomorphism from the finite free $D(n)\to R$ to $X\to R$.

**Comment by felixwellen** [comment]
I cannot reproduce my last statement, but what we have is that all maps $f:X\to R$ factor over an infinitesimal disk around $f(\ast)$ of fixed order.

**Comment by felixwellen** [comment]
Another approach is to look at an embedding $f:\mathrm{Spec}(A)\hookrightarrow \mathbb{A}^n$ which merely exists for affine $X=\mathrm{Spec}(A)$. Then, if $X$ is projective, we can look again at $g:\mathrm{Spec}(A)\hookrightarrow\mathbb{A}^n\hookrightarrow \mathbb{P}^n$ given by $x\mapsto [f_1(x):\dots:f_n(x):1]$. Then $U([x_1:\dots:x_{n+1}]):=(y:\mathrm{Spec}(A))\to \bigvee_i x_{n+1}\cdot f(y)\neq x_i$ defines again an open which contains the hyperplane $x_{n+1}=0$ and has empty intersection with $f(\mathrm{Spec}(A))$. 
One can hope that it is possible to exploit this situation algebraically, to show something about $\mathrm{Spec}(A)$.

**Comment by dwarn** [comment]
It seems one can use the above idea to show that $A$ is finitely generated as an $R$-module. I believe a finitely presented algebra that is finitely generated as a module is also finitely presented as a module (?), so $A$ should at least be finitely presented as an $R$-module. One can still hope that it is finite free. So far the argument has not used the assumption that $\mathrm{Spec} A$ is projective except to conclude that it is compact.

We argue as follows. The open $U \subseteq \mathbb P^n$ defined above is given by a homogeneous ideal $(Q_1, \ldots, Q_r)$ in $R[x_1,\ldots,x_{n+1}]$. Write $\mathrm{Spec}(A) = V(P_1,\ldots,P_k)$ with $P_i$ polynomials in $n$ variables. We know two things about $U$: $U \cap \mathbb A^n = \mathbb A^n \setminus \mathrm{Spec}(A) = D(P_1,\ldots, P_k)$, and $U \cup \mathbb A^n = \mathbb P^n$. The first equation tells us that $Q_i(x_1,\ldots,x_n,1) \in \sqrt{(P_1,\ldots,P_k)}$. Replacing $Q_i$ with some power of it if necessary, we may assume $Q_i(x_1,\ldots,x_n,1) \in (P_1,\ldots,P_k)$ for all $i$. The second equation means that
$D(Q_1,\ldots,Q_r,x_{n+1}) = \mathbb P^n$, which means that
$x_i \in \sqrt{(Q_1,\ldots,Q_r,x_{n+1})}$ for (in particular) $i$ between $1$ and $n$. Write $x_i^{M_i} = f + Rx_{n+1}$ where $f \in (Q_1,\ldots,Q_r)$ with $f$ and $R$ homogeneous, of degrees $M_i$ and $M_i-1$. Now write $g \coloneqq f(x_1,\ldots,x_n,1)$, so $g \in (P_1,\ldots,P_k)$. We have $x_i^{M_i} = f(x_1,\ldots,x_n,0)$, so $g$ is $x_i^{M_i}$ plus a polynomial of degree at most $M_i-1$. This means that $A = R[x_1,\ldots,x_n]/(P_1,\ldots,P_k)$ is spanned by monomials whose $x_i$-degree is at most $M_i$. So, we have a generating family of size $M_1\cdots M_n$.

**Comment by iblech** [comment]
Very interesting discussion!

Regarding David's hope about finite presentation: This is indeed correct, a reference is Tag 0564 in the Stacks Project; at some point I convinced myself that the proof offered there is constructive (or easily constructivized).

**Comment by felixwellen** [comment]
Nice! I guess we could also phrase it like this:
$\mathrm{Spec}(A)$ is compact/proper iff it is projective iff $A$ is an fp $R$-module. 

Since only compactness was used above.

**Comment by coquand** [comment]
I think we use something similar in the projective paper.

 We have U open of P1 = U0 \/ U1 and U contains [1:0]

 U0    [r:s] with s <> 0
 U1    [r:s] with s <> 0

 Hence U /\ U1 contains [1:0]

 So we have g(s) with g(0) <> 0 and D(g) subset U /\ U1 

 We can assume g(0) = 1.

 Then F = X X^d g(1/X) if d formal degree of g will be such that D(F) subset U0 /\ U /\ U1 and F monic.

 So Im a and D(F) are disjoint.
